src/brewvoid.h

```diff
@@ -23,22 +23,21 @@ void checkbrewswitch() {
         #if (PIN_BREWSWITCH > 0)
             int reading = digitalRead(PIN_BREWSWITCH);
 
-            if (reading != brewswitchTrigger) {
-                // reset the debouncing timer
+            if (reading != lastButtonStateBrew) {
+                // restart the debouncing timer
                 lastDebounceTimeBrewTrigger = millis();
+                // set new button state 
+                lastButtonStateBrew = reading;
             }
-
-            if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
+            else if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
                 // whatever the reading is at, it's been there for longer than the debounce
                 // delay, so take it as the actual current state:
-
-                // if the button state has changed:
-                if (reading != buttonStateBrewTrigger) {
-                    buttonStateBrewTrigger = reading;
+                
+                if (brewswitchTrigger != reading)
```

Comment:
Could you put the closing bracket inline with the if statement? It's the code style we generally use

Reply:
Can't be. Its the mechanical switch, converted to a push button:

Code is executed if:
#if BREWSWITCHTYPE == 2
        #if (PIN_BREWSWITCH > 0)

Config file:
#define BREWSWITCHTYPE 2           // 0 = no switch connected, 1 = normal switch, 2 = trigger switch

I tested with the brew switch, no issues at all.

Reply:
Ah yeah, the code path wouldn't make sense with software brew detection - the trigger switch is detected already. Then it should be fine